flink-kubernetes/src/test/java/org/apache/flink/kubernetes/highavailability/KubernetesTestFixture.java

```diff
@@ -0,0 +1,236 @@
+
+    private final CompletableFuture<KubernetesLeaderElector.LeaderCallbackHandler>
+            leaderCallbackHandlerFuture = new CompletableFuture<>();
+
+    private final FlinkKubeClient flinkKubeClient;
+
+    private final KubernetesConfigMapSharedWatcher configMapSharedWatcher;
+
+    KubernetesTestFixture(String clusterId, String leaderConfigmapName, String lockIdentity) {
+        this.leaderConfigmapName = leaderConfigmapName;
+        this.lockIdentity = lockIdentity;
+        configuration = new Configuration();
+        configuration.setString(KubernetesConfigOptions.CLUSTER_ID, clusterId);
+
+        flinkKubeClient = createFlinkKubeClient();
+        configMapSharedWatcher =
+                flinkKubeClient.createConfigMapSharedWatcher(
+                        KubernetesUtils.getConfigMapLabels(
+                                clusterId, LABEL_CONFIGMAP_TYPE_HIGH_AVAILABILITY));
+    }
+
+    void close() {
+        configMapSharedWatcher.close();
+    }
+
+    FlinkKubeClient getFlinkKubeClient() {
+        return flinkKubeClient;
+    }
+
+    CompletableFuture<Void> getCloseKubeClientFuture() {
+        return closeKubeClientFuture;
+    }
+
+    CompletableFuture<Map<String, String>> getDeleteConfigMapByLabelsFuture() {
+        return deleteConfigMapByLabelsFuture;
+    }
+
+    KubernetesConfigMapSharedWatcher getConfigMapSharedWatcher() {
+        return configMapSharedWatcher;
+    }
+
+    Configuration getConfiguration() {
+        return configuration;
+    }
+
+    KubernetesConfigMap getLeaderConfigMap() {
+        final Optional<KubernetesConfigMap> configMapOpt =
+                flinkKubeClient.getConfigMap(leaderConfigmapName);
+        assertThat(configMapOpt.isPresent(), is(true));
+        return configMapOpt.get();
+    }
+
+    // Use the leader callback to manually grant leadership
+    void leaderCallbackGrantLeadership() throws Exception {
+        createLeaderConfigMap();
+        getLeaderCallback().isLeader();
+    }
+
+    FlinkKubeClient.WatchCallbackHandler<KubernetesConfigMap> getLeaderElectionConfigMapCallback()
+            throws Exception {
+        assertThat(configMapCallbackFutures.size(), is(greaterThanOrEqualTo(1)));
+        return configMapCallbackFutures.get(0).get(TIMEOUT, TimeUnit.MILLISECONDS);
+    }
+
+    FlinkKubeClient.WatchCallbackHandler<KubernetesConfigMap> getLeaderRetrievalConfigMapCallback()
+            throws Exception {
+        assertThat(configMapCallbackFutures.size(), is(2));
+        return configMapCallbackFutures.get(1).get(TIMEOUT, TimeUnit.MILLISECONDS);
+    }
+
+    KubernetesLeaderElector.LeaderCallbackHandler getLeaderCallback() throws Exception {
+        return leaderCallbackHandlerFuture.get(TIMEOUT, TimeUnit.MILLISECONDS);
+    }
+
+    private FlinkKubeClient createFlinkKubeClient() {
+        return createFlinkKubeClientBuilder().build();
+    }
+
+    TestingFlinkKubeClient.Builder createFlinkKubeClientBuilder() {
```

Comment:
I rather would like to not do this because I don't see an immediate benefit. If this should be necessary in the future, then let's spend the time on it.